src/shared/modules/loader/components/loader.component.ts

```diff
@@ -0,0 +1,23 @@
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { LoaderService } from '../services/loader.service';
+import { LoaderState } from '../loader.model';
+@Component({
+  selector: 'app-loader',
+  templateUrl: './loader.component.html',
+  styleUrls: ['./loader.component.css']
+})
+export class LoaderComponent implements OnInit, OnDestroy {
+  show = false;
+  private subscription: Subscription;
+  constructor(private loaderService: LoaderService) { }
+  ngOnInit() {
+    this.subscription = this.loaderService.loaderState
+    .subscribe((state: LoaderState) => {
```

Comment:
can u use the async pipe here rather then subscribing it will save few lines of code

Reply:
Agree with Franco here